framework/decode/vulkan_ascii_consumer_base.h

```diff
@@ -109,22 +111,52 @@ class VulkanAsciiConsumerBase : public VulkanConsumer
 
   protected:
     template <typename ToStringFunctionType>
-    inline void WriteApiCallToFile(const ApiCallInfo&   call_info,
-                                   const std::string&   functionName,
-                                   util::ToStringFlags  toStringFlags,
-                                   uint32_t&            tabCount,
-                                   uint32_t             tabSize,
-                                   ToStringFunctionType toStringFunction)
+    inline void WriteApiCallToFile(const ApiCallInfo& call_info,
+                                   const std::string& functionName,
+                                   /// @todo Remove toStringFlags, tabCount, tabSize
+                                   util::ToStringFlags        toStringFlags,
+                                   const uint32_t&            tabCount,
+                                   const uint32_t             tabSize,
+                                   const ToStringFunctionType toStringFunction,
+                                   /// The formatted return string value excluding trailing comma
+                                   const char* const return_val = nullptr)
     {
-        using namespace util;
-        fprintf(file_, "%s\n", (call_info.index ? "," : ""));
-        fprintf(file_, "\"[%s]%s\":", std::to_string(call_info.index).c_str(), functionName.c_str());
-        fprintf(file_, "%s", GetWhitespaceString(toStringFlags).c_str());
```

Comment:
There is no formatting applied. This and the next line should never have been `fprintf()`s. We just want to copy the strings into the file output buffer directly. Hence change to `fputs()`.

Reply:
👍